adcc/ExcitedStates.py

```diff
@@ -657,6 +659,48 @@ def describe_amplitudes(self, tolerance=0.01, index_format=None):
                 ret += separator
         return ret[:-1]
 
+    def to_dataframe(self, export_vectors=True):
+        """
+        Exports the ExcitedStates object as :class:`pandas.DataFrame`.
+
+        Parameters
+        ----------
+        export_vectors: bool, optional
+            Whether to export vector quantities (transition moments)
+        """
+        propkeys = extract_excitation_property_keys(self)
+        propkeys.extend(self.excitation_energy_corrections.keys())
+        data = {
+            "excitation": np.arange(0, self.size, dtype=int),
+            "kind": np.tile(self.kind, self.size)
+        }
+        # fill the dict with data, sort out stuff that we cannot represent
+        # in a table
+        from adcc import Tensor, AmplitudeVector, OneParticleOperator
+        unacceptable_types = [AmplitudeVector, OneParticleOperator, Tensor]
+        for key in propkeys:
+            d = getattr(self, key)
+            if isinstance(d, list):
+                if any(type(e) in unacceptable_types for e in d):
+                    continue
+                data[key] = d
+            elif isinstance(d, np.ndarray):
+                if d.ndim == 1:
+                    data[key] = d
+                elif d.ndim == 2 and d.shape[1] == 3 and export_vectors:
+                    component = ["x", "y", "z"]
+                    # TODO: ugly, but works...
+                    for i, p in enumerate(component):
```

Comment:
Completely fine, I think. I'd even do it in one line.

Reply:
Done.